File: server_client_java_rpc_protocol/JavaSocketsFestival_protocol_rpc/Java11ChatSockets/ChatPersistence/src/main/java/festival/persistence/repository/DBRepoArtist.java
package festival.persistence.repository;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import festival.model.Artist;
import festival.model.Identifier;
import festival.model.Persoana;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Optional;
import java.util.Properties;

public class DBRepoArtist implements IDBRepoArtist{
    private JdbcUtils dbUtils;
    private static final Logger logger= LogManager.getLogger();


    public DBRepoArtist(Properties props) {
        logger.info("Initializing DBRepoArtist with properties: {} ",props);
        dbUtils=new JdbcUtils(props);
    }


    @Override
    public Iterable findAll() {

        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        List<Artist> artisti = new ArrayList<>();
        try (PreparedStatement preStmt = con.prepareStatement(
                "SELECT P.nume, P.prenume, P.IDPerosana, A.IDArtist\n" +
                "FROM persoana P  \n" +
                "JOIN artist A ON A.IDPersoana=P.IDPerosana \n"))
        {
            try (ResultSet result = preStmt.executeQuery()) {
                while (result.next()) {
                    String nume = result.getString("nume");
                    String prenume = result.getString("prenume");

                    int id_persoana = result.getInt("IDArtist");

                    Artist artist= new Artist((long) id_persoana,nume,prenume);
//                    artist.setNume(nume);
//                    artist.setPrenume(prenume);
                    artisti.add(artist);

                }
            }
        } catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit(artisti);
        return artisti;

    }

    @Override
    public Optional save(Identifier entity) {
        return Optional.empty();
    }

    @Override
    public Optional findOne(Object longID) {
        return Optional.empty();
    }

    @Override
    public Optional delete(Object aLong) {
        return Optional.empty();
    }


}

File: server_client_C_sharp/mpp_c_app1/FestivalPersistence/DBRepoArtist.cs

using log4net;
using System;
using System.Collections.Generic;
using System.Data.SQLite;
using FestivalModel;

namespace FestivalPersistence
{

    public interface IDBRepoArtist : IRepository<long, Artist>
    {
    }

    public class DBRepoArtist : IDBRepoArtist
    {
        private SQLiteConnection connection;
        private static readonly ILog logger = LogManager.GetLogger("DBRepoArtist");

        public DBRepoArtist(string connectionString)
        {
            logger.Info("Initializing DBRepoArtist...");
            connection = new SQLiteConnection(connectionString);
        }

        public IEnumerable<Artist> findAll()
        {
            logger.InfoFormat("findAll method called.");
            List<Artist> artisti = new List<Artist>();
            using (SQLiteCommand command = new SQLiteCommand("SELECT P.nume, P.prenume, A.IDArtist " +
                                                             "FROM persoana P " +
                                                             "JOIN artist A ON A.IDPersoana=P.IDPerosana", connection))
            {
                connection.Open();
                using (SQLiteDataReader reader = command.ExecuteReader())
                {
                    while (reader.Read())
                    {
                        string nume = reader.GetString(0);
                        string prenume = reader.GetString(1);
                        long id = reader.GetInt64(2);

                        Artist artist = new Artist(id, nume, prenume);

                        artisti.Add(artist);
                    }
                }

                connection.Close();
            }

            logger.InfoFormat("Found {0} artists", artisti.Count);
            return artisti;
        }

        public Artist findOne(long id)
        {
            logger.InfoFormat("findOne method called for id: {0}", id);
            using (SQLiteCommand command = new SQLiteCommand("SELECT P.nume, P.prenume, A.IDArtist " +
                                                             "FROM persoana P " +
                                                             "JOIN artist A ON A.IDPersoana=P.IDPerosana " +
                                                             "WHERE P.IDPerosana = @id", connection))
            {
                command.Parameters.AddWithValue("@id", id);
                connection.Open();
                using (SQLiteDataReader reader = command.ExecuteReader())
                {
                    if (reader.Read())
                    {
                        string nume = reader.GetString(0);
                        string prenume = reader.GetString(1);
                        connection.Close();
                        return new Artist(id, nume, prenume);
                    }
                }

                connection.Close();
            }

            return null;
        }

        public void save(Artist entity)
        {
            logger.InfoFormat("save method called for artist with id: {0}", entity.Id);
            using (SQLiteCommand command =
                   new SQLiteCommand("INSERT INTO persoana (nume, prenume) VALUES ( @nume, @prenume)", connection))
            {
                command.Parameters.AddWithValue("@nume", entity.Nume);
                command.Parameters.AddWithValue("@prenume", entity.Prenume);
                connection.Open();
                command.ExecuteNonQuery();
                connection.Close();
            }
        }

        public void delete(long id)
        {
            logger.InfoFormat("delete method called for id: {0}", id);
            using (SQLiteCommand command = new SQLiteCommand("DELETE FROM persoana WHERE id = @id", connection))
            {
                command.Parameters.AddWithValue("@id", id);
                connection.Open();
                command.ExecuteNonQuery();
                connection.Close();
            }
        }
    }
}